core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -632,17 +630,15 @@ public boolean process()
         }
 
         @Override
-        public GroupByIdBlock getResult()
+        public int[] getResult()
         {
             checkState(processFinished);
             checkState(!resultProduced);
             resultProduced = true;
 
-            return new GroupByIdBlock(
-                    nextGroupId,
-                    RunLengthEncodedBlock.create(
-                            BIGINT.createFixedSizeBlockBuilder(1).writeLong(groupId).build(),
-                            block.getPositionCount()));
+            int[] result = new int[block.getPositionCount()];
```

Comment:
Would it be beneficial to handle this case better?
We could still have `GroupIdBLock` that has two states, int[] and single value and count similar to `SelectedPositions` and then do 
```
if (groupIdsBlock.isArray()) {
  // handle array
} else {
  // handle rle
}
``` in the `Accumulator`

Reply:
Maybe. I'm not sure the complexity will result in better performance in the 99% case where we don't have RLEs.  My thought is we switch to `int[]` and establish a new base line.  Then in follow up work we can investigate if adding wrapper with RLE support helps.